packages/block-editor/src/components/link-control/link-preview.js

```diff
@@ -59,6 +59,22 @@ export default function LinkPreview( {
 		! isEmptyURL &&
 		stripHTML( richData?.title || value?.title || displayURL );
 
+	/**
+	 * Filters the title for display. Removes the protocol and www prefix.
+	 *
+	 * @param {string} title - The title to be filtered.
+	 *
+	 * @return {string} The filtered title.
+	 */
+	function filterTitleForDisplay( title ) {
+		return title
+			.replace( /^[a-z\-.\+]+[0-9]*:(\/\/)?/i, '' )
```

Comment:
Worth adding a comment that this is derived from `@wordpress/url`'s `isValidProtocol()` utility.

Reply:
Since this regex is added in `filterURLForDisplay()` so I have added this function name in the inline comment.